service/storage_proxy.cc

```diff
@@ -409,6 +410,12 @@ class storage_proxy::remote {
             tracing::trace(trace_state_ptr, "Message received from /{}", src_addr.addr);
         }
 
+        const auto fence = fence_opt.value_or(fencing_token{});
+        if (auto stale = _sp.apply_fence(fence, src_addr.addr)) {
+            co_return co_await encode_replica_exception_for_rpc<replica::exception_variant>(_sp.features(),
+                make_exception_future<>(std::move(stale)));
+        }
+
```

Comment:
Checking at the beginning is just an optimization, right?

Reply:
Yes, if we already know that the current topology is stale, we can return early.